### PR TITLE
Adds zen layout for 333fred

### PR DESCRIPTION
**Pull request by 333fred**
Adds support for my Zen layout, and some basic rgb support for layer-changing. @drashna for review and merge.

### CODE REVIEW

File: keyboards/zen/keymaps/333fred/config.h [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#ifndef CONFIG_USER_H
```
Comment (drashna) [inline keyboards/zen/keymaps/333fred/config.h]: Could you replace the boiler plate with `#pragma once`? 

Eg, remove this: 
```h
#ifndef CONFIG_USER_H
#define CONFIG_USER_H

#include "../../config.h"
```
And the `#endif` at the end with just `#pragma once`

File: keyboards/zen/keymaps/333fred/rules.mk [hunk 2]
```diff
@@ -0,0 +1,12 @@
+RGBLIGHT_ENABLE = yes
+BACKLIGHT_ENABLE = no
+NKRO_ENABLE = yes
+KEY_LOCK_ENABLE = yes
+TAP_DANCE_ENABLE = yes
+CONSOLE_ENABLE = no
+
+ifndef QUANTUM_DIR
```
Comment (drashna) [inline keyboards/zen/keymaps/333fred/rules.mk]: If you could remove this block?  It's a holdover from the old makefile system, and not needed anymore.

**Comment by 333fred** [comment]
@drashna addressed feedback.

**Comment by drashna** [comment]
Awesome, thanks!

**Pull request merged**